### PR TITLE
http2: minor PersistentConnection cleanups

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

**Review by raboof — APPROVED** [review]
:+1:

File: akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala [hunk 1]
```diff
@@ -125,10 +125,10 @@ private[http2] object PersistentConnection {
           requestOut.fail(new RuntimeException("connection broken"))
 
           if (connectsLeft.contains(0)) {
-            failStage(cause)
+            failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts"))
```
Comment (ignasi35) [inline akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala:128]:
```suggestion
            failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts"), cause)
```

Keep the cause around? Or maybe add the `cause.getMessage` in a log? (like in line 131)

File: akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala [hunk 2]
```diff
@@ -125,10 +125,10 @@ private[http2] object PersistentConnection {
...
           } else {
             setHandler(requestIn, Unconnected)
-            log.info(s"failed, trying to connect again: ${cause.getMessage}${connectsLeft.map(n => s" ($n left)").getOrElse("")}")
+            log.info(s"Connection attempt failed, trying to connect again: ${cause.getMessage}${connectsLeft.map(n => s" ($n left)").getOrElse("")}")
```
Comment (ignasi35) [inline akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala:131]:
```suggestion
            log.info(s"Connection attempt failed: ${cause.getMessage}. Trying to connect again${connectsLeft.map(n => s" ($n attempts left)").getOrElse("")}.")
```

Comment (ignasi35) [reply akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala:131]: 👆🏼 nitpick

**Review by ignasi35 — APPROVED** [review]
LGTM with a couple of minor nitpicks

**Pull request merged**